test/functional/core/path_spec.lua

```diff
@@ -103,6 +105,36 @@ describe('file search', function()
     eq('filename_with_unicode_ααα', eval('expand("%:t")'))
   end)
 
+  it('finds "file:[/|///]" URI on the local filesystem #24032', function()
+    local function test_gf(input, expected)
+      command('%delete')
+      insert(input)
+      command('norm! 0')
+      feed('gf')
+      eq(expected, eval('expand("%:p")'))
+    end
+
+    test_gf("file:/a", "/a")
+    test_gf("file:/foo/bar/slash.txt", "/foo/bar/slash.txt")
+    test_gf("file:///foo/bar/3slashes.txt", "///foo/bar/3slashes.txt")
+    test_gf("file:/c:/test/window/slash.txt", "/c:/test/window/slash.txt")
+    test_gf("file:///c:/test/window/3slash.txt", "///c:/test/window/3slash.txt")
+    test_gf("http://foo/bar/file:/with/url", "http://foo/bar/file:/with/url")
+
+    -- Don't handle `gf file://`
+    command('%delete')
+    insert("file://host/foo/bar/2slashes.txt")
+    command('norm! 0')
+    feed('gf')
+    neq("//host/foo/bar/2slashes.txt", eval('expand("%:p")'))
```

Comment:
try to use positive confirmation (eq instead of neq), it results in must better failure messages and avoids false successes.